core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -634,6 +636,10 @@ boolean isPreserveAliases() {
                 if (!noIndicesAllowedOrMatches(options, matches)) {
                     throw infe(expression);
                 }
+
+                if (expression.contains("*")) {
```

Comment:
How about the ordering thing that I mentioned before? Should `-test1,test*` resolve to the same indices as `test*,-test1`? It doesn't now right?